### PR TITLE
style: Auto format code using prettier before commit

### PR DESCRIPTION
**Pull request by mrchief**
## What does it do?

Adds prettierrc and pre commit hooks to format code automatically. Cleans up eslintrc and switched to react-app lint rules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,13 +26,12 @@
     "build": "rimraf dist/**/* && cross-env NODE_ENV=production webpack --config webpack.config.js --bail --mode=production",
```
Comment (ellinge) [inline package.json]: Checked if we could remove any more dependencies with depcheck after changing linting. This is not 100% but some hints at least. As you can see some of these are false positives. Either way the tool reports:

Unused devDependencies
* @babel/plugin-syntax-jsx
* @babel/preset-stage-3
* @commitlint/cli
* @commitlint/config-conventional
* babel-plugin-transform-react-remove-prop-types
* conventional-changelog-cli
* eslint-config-prettier
* husky
* ignore-styles
* jsdom
* jsdom-global
* lint-staged
* postcss
* prettier-stylelint
* raw-loader
* style-loader
* stylelint-config-standard
* url-loader
* webpack-cli
* why-did-you-update

Comment (mrchief) [reply package.json]: > Checked if we could remove any more dependencies with depcheck after changing linting.

Glad I'm not the only dependency-averse person out there :)

I've given up on depcheck years ago. I did give it another try yesterday but nothing has changed. That said, I did check the list and I know where each one of those deps are used, so you're right about them being false positives.

**Comment by mrchief** [comment]
@ellinge I'm going to hold on to this PR till we merge the older ones first. That way, we have to solve conflicts in one branch as opposed to solving in 3 otherwise.

**Comment by mrchief** [comment]
Maybe we should merge this now? I don't think I can ever catch up to you :)

**Comment by ellinge** [comment]
Sure! Hoped it was the radioSelect @ first from the mail notification... 🥈

**Comment by mrchief** [comment]
> Sure! Hoped it was the radioSelect @ first from the mail notification... 🥈

Alright, alright! We'll do this after radio select then 😃 

> Seems to be some build errors though.

We need to pretty things after every merge 🎀

**Pull request merged**